core/services/head_tracker.go

```diff
@@ -600,14 +601,20 @@ func (ht *HeadTracker) unsubscribeFromHead() error {
 }
 
 func (ht *HeadTracker) setHighestSeenHeadFromDB() error {
-	head, err := ht.store.LastHead(context.Background())
+	head, err := ht.HighestSeenHeadFromDB()
 	if err != nil {
 		return err
 	}
 	ht.highestSeenHead = head
 	return nil
 }
 
+func (ht *HeadTracker) HighestSeenHeadFromDB() (*models.Head, error) {
+	ctx, cancel := utils.ContextFromChan(ht.done)
```

Comment:
this should probably be a combined context that includes `postgres.DefaultQueryCtx()`

Reply:
done 👍🏼